Automattic/pocket-casts-android
# End of Year: Add stories datasource + gestures

| 📘 Project: #410 |
|:---:|

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/389

# Description

This PR adds a data source to populate the Stories view with stories and also adds a few gestures.
Currently, it just updates the background color of the StoryView. Different story views will be added in future PRs.

https://user-images.githubusercontent.com/1405144/197767780-76d7537a-fac9-40a6-a361-f2222a04e237.mp4

## To test

1. Set `END_OF_YEAR_ENABLED` feature flag to `true` in `base.gradle`
2. Run the app
3. When the prompt appears, tap "View My 2022"
4. ✅ Check that each story is shown for 2s and that the story indicator (the rectangles at the top) are reflecting the state
5. ✅ Tap and hold, check that the current story is "paused". The rectangle stop progressing and the story doesn't change
6. ✅ While seeing the green story, tap the left portion of the screen, you should go back to the magenta one
7. ✅ While seeing the magenta story, tap the right portion of the screen, you should go to the next story (green)


**Loading and error**

2. Return an empty list from `EndOfYearStoriesDataSource.loadStories`
3. Run the app
4. When the prompt appears, tap "View My 2022"
5. ✅ Check that loading indicator appears while stories are loading (I added a delay to test this)
6. Wait for a sec
7. ✅ Notice that error message is displayed when no stories are found.


https://user-images.githubusercontent.com/1405144/197771060-b651452f-38ac-4c04-b42d-aab17980e6a9.mp4



# Checklist

- Should this change be included in the release notes? If yes, please add a line in CHANGELOG.md
- Have you tested accessibility with TalkBack?
- Does the change work with a large display font?
- Are all the strings localized?
